src/crimson/os/futurized_store.h

```diff
@@ -45,12 +48,39 @@ class FuturizedStore {
 
   using mount_ertr = crimson::errorator<crimson::stateful_ec>;
   virtual mount_ertr::future<> mount() = 0;
+
   virtual seastar::future<> umount() = 0;
 
   using mkfs_ertr = crimson::errorator<crimson::stateful_ec>;
   virtual mkfs_ertr::future<> mkfs(uuid_d new_osd_fsid) = 0;
+
   virtual seastar::future<store_statfs_t> stat() const = 0;
 
+  virtual uuid_d get_fsid() const  = 0;
+
+  virtual seastar::future<> write_meta(const std::string& key,
+				       const std::string& value) = 0;
+
+  virtual FuturizedShardStore& get_sharded_store() = 0;
+
+  virtual seastar::future<std::tuple<int, std::string>> read_meta(
+    const std::string& key) = 0;
+
+  using coll_core_t = std::pair<coll_t, core_id_t>;
+  virtual seastar::future<std::vector<coll_core_t>> list_collections() = 0;
+
+protected:
+  core_id_t primary_core;
+};
+
+class FuturizedShardStore {
```

Comment:
This is perhaps a minor point, but FuturizedStore->FuturizedShardStore is confusing and redundant.  If we nest this within FuturizedStore, we can simply name it Shard or FuturizedStore::Shard.  This would then lead to CyanStore::Shard, AlienStore::Shard, and Seastore::Shard.  At the very least, I'd prefer FuturizedStoreShard to FuturizedShardStore (ditto for the other three).